### PR TITLE
Phil branch

### PR DESCRIPTION
**Pull request by phil-fradkin**
- add default flag for dropping ionic compounds
- Default now multi class classification
- Remove duplicates from cpdb so no duplciated smiles (blah)
- add lr decay
- log validation loss
- add script to install DIG dependencies

### CODE REVIEW

File: src/arguments.py [hunk 1]
```diff
@@ -49,15 +49,17 @@ def get_args():
     parser.add_argument("--conv_num_kernels", type=int, default=32)
     parser.add_argument("--conv_num_layers", type=int, default=3)
     # training stuff
-    parser.add_argument("--random_seed", type=int, default=420420420)
+    parser.add_argument("--random_seed", type=int, default=np.random.randint(100000))
```
Comment (adamoyoung) [inline src/arguments.py:52]: what's the motivation for sampling a diff random seed by default, for every run?

File: src/data_loader.py [hunk 2]
```diff
@@ -236,30 +254,36 @@ def process(self):
...
-                dgllife.utils.smiles_to_bigraph(
-                    x,
-                    add_self_loop=True,
-                    node_featurizer=node_featurizer,
-                    edge_featurizer=edge_featurizer,
-                    canonical_atom_order=False,
-                ) for x in self.df['smiles'].values
-            ]
-            self.df["graph"] = bigraphs
+            bigraphs = []
```
Comment (adamoyoung) [inline src/data_loader.py:257]: do we need to undo this change? is there a problem? I like filtering in the df because it makes it easier to see how many we've filtered (like you only remove them at the last step). this has the benefit of also allowing us to add additional filters really easily

File: src/train.py [hunk 3]
```diff
@@ -102,15 +108,22 @@ def main():
...
+    optimizer = Adam(model.parameters(), lr=args['lr'], weight_decay=args['lr_decay'])
 
     if args["use_wandb"]:
         es_filename = os.path.join(wandb.run.dir,"early_stopping")
     else:
         es_filename = "/tmp/es.pth"
+
+    if args['metric'] == 'validation_loss':
+        # ugly hack because we're using a different loss
+        metric_for_stopper = 'rmse'
```
Comment (adamoyoung) [inline src/train.py:120]: ye i kinda dislike the DGL stopper because it assumes a lot, but this hack works

**Review by adamoyoung — COMMENTED** [review]
I added some comments

File: src/arguments.py (hunk 1 above)
Comment (phil-fradkin) [reply src/arguments.py:52]: I read a twitter thread that it's possible to overfit to a single random seed https://twitter.com/rishabh16_/status/1394613798242115587

I'm not 100% convinced that it's possible but there doesn't seem to be an obvious downside. If one hyperparam choice is better than the other it should be consistent across random seeds.

File: src/data_loader.py (hunk 2 above)
Comment (phil-fradkin) [reply src/data_loader.py:257]: yeah that's fair this is the stack trace that I get
```
Traceback (most recent call last):
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/site-packages/ipdb/__main__.py", line 301, in main
    pdb._runscript(mainpyfile)
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/pdb.py", line 1568, in _runscript
    self.run(statement)
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/bdb.py", line 578, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "/Users/phil/Documents/01_projects/03_learning_molecular_representation/CONCERTO/src/data_loader.py", line 1, in <module>
    from rdkit import Chem
  File "/Users/phil/Documents/01_projects/03_learning_molecular_representation/CONCERTO/src/data_loader.py", line 456, in test_ds
    train_data, _ = get_datasets(args)
  File "/Users/phil/Documents/01_projects/03_learning_molecular_representation/CONCERTO/src/data_loader.py", line 45, in get_datasets
    **kwargs
  File "/Users/phil/Documents/01_projects/03_learning_molecular_representation/CONCERTO/src/data_loader.py", line 219, in __init__
    self.process()
  File "/Users/phil/Documents/01_projects/03_learning_molecular_representation/CONCERTO/src/data_loader.py", line 248, in process
    self.df["graph"] = bigraphs
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/site-packages/pandas/core/frame.py", line 3163, in __setitem__
    self._set_item(key, value)
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/site-packages/pandas/core/frame.py", line 3242, in _set_item
    value = self._sanitize_column(key, value)
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/site-packages/pandas/core/frame.py", line 3902, in _sanitize_column
    value = maybe_convert_platform(value)
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/site-packages/pandas/core/dtypes/cast.py", line 112, in maybe_convert_platform
    values = construct_1d_object_array_from_listlike(values)
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/site-packages/pandas/core/dtypes/cast.py", line 1638, in construct_1d_object_array_from_listlike
    result[:] = values
  File "/Users/phil/anaconda3/envs/lm_rep/lib/python3.7/site-packages/dgl/heterograph.py", line 2152, in __getitem__
    raise DGLError('Invalid key "{}". Must be one of the edge types.'.format(orig_key))
dgl._ffi.base.DGLError: Invalid key "0". Must be one of the edge types.
Uncaught exception. Entering post mortem debugging
Running 'cont' or 'step' will restart the program
```

File: src/arguments.py (hunk 1 above)
Comment (adamoyoung) [reply src/arguments.py:52]: cool. Sticking to one seed is nice for development, but I don't think there's a consensus on what the best practice is for hyperopt seeds (I checked out the thread and everyone was arguing!). The real best practice (if you have the resources/time) is to run multiple independent hyperopts with different random seeds. I don't feel strongly one way or the other so I think this is fine!

Comment (adamoyoung) [reply src/arguments.py:52]: it's basically like: do you want a biased hyperopt run, or a higher variance hyperopt run? both can be bad. hard to tell a priori

Comment (phil-fradkin) [reply src/arguments.py:52]: That's a good take away! In the end we're just looking for a model that works well and whether it contains the elusive "lottery ticket" in its weights or is a great combination of hyperparams making it easier for the model to learn the specific task at hand our only real way to currently identify it is through it's validation performance.

File: src/data_loader.py (hunk 2 above)
Comment (adamoyoung) [reply src/data_loader.py:257]: I am spooked and bewildered

**Review by adamoyoung — APPROVED** [review]
